vsphere/check.py

```diff
@@ -924,6 +957,383 @@ def collect_metrics(self, instance):
+                    add_label_pair(labels, "guestId", c.config.guestId)
+                    add_label_pair(labels, "guestFullName", c.config.guestFullName)
+                    add_label_pair(labels, "numCPU", c.config.hardware.numCPU)
+                    add_label_pair(labels, "memoryMB", c.config.hardware.memoryMB)
+                topology_tags["labels"] = labels
+                obj_list.append(dict(mor_type="vm", mor=c, hostname=hostname, topo_tags=topology_tags))
+
+        return obj_list
+
+    def _vsphere_datacenters(self, content, domain="Unspecified", regexes=None, include_only_marked=False):
+
+        def add_label_pair(label_list, key, value):
+            label_list.append("{0}:{1}".format(key, value))
+
+        obj_list = []
+        container = content.viewManager.CreateContainerView(
+            content.rootFolder,
+            [RESOURCE_TYPE_MAP["datacenter"]],
+            True)
+
+        for c in container.view:
+            topology_tags = {}
+            labels = []
+            if not self._is_excluded(c, regexes, include_only_marked):
+                hostname = c.name
+                if isinstance(c, vim.Datacenter):
+                    datastores = []
+                    for datastore in c.datastore:
+                        datastores.append(datastore.name)
+                    topology_tags["topo_type"] = VSPHERE_COMPONENT_TYPE.DATACENTER
+                    topology_tags["datastores"] = datastores
+                    topology_tags["name"] = c.name
+                    topology_tags["id"] = c._moId
+                    topology_tags["layer"] = TOPOLOGY_LAYERS.DATACENTER
+                    topology_tags["domain"] = domain
+
+                    computeresources = []
+                    clustercomputeresources = []
+
+                    for computeres in c.hostFolder.childEntity:
+                        if isinstance(computeres, vim.ComputeResource):
+                            computeresources.append(computeres.name)
+                        elif isinstance(computeres, vim.CloudComputeResource):
+                            clustercomputeresources.append(computeres.name)
+
+                    topology_tags["computeresources"] = computeresources
+                    topology_tags["clustercomputeresources"] = clustercomputeresources
+
+                    add_label_pair(labels, "name", topology_tags["name"])
+                    hostname = None
+                topology_tags["labels"] = labels
+                obj_list.append(dict(mor_type="datacenter", mor=c, hostname=hostname, topo_tags=topology_tags))
+
+        return obj_list
+
+    def _vsphere_datastores(self, content, domain="Unspecified", regexes=None, include_only_marked=False):
+
+        def add_label_pair(label_list, key, value):
+            label_list.append("{0}:{1}".format(key, value))
+
+        obj_list = []
+        container = content.viewManager.CreateContainerView(
```

Comment:
duplicate code, maybe split off to a method accepting a vimtype and returing the container.view?

Reply:
addressed